src/hotspot/share/ci/ciField.cpp

```diff
@@ -216,6 +216,9 @@ ciField::ciField(fieldDescriptor *fd) :
 static bool trust_final_non_static_fields(ciInstanceKlass* holder) {
   if (holder == nullptr)
     return false;
+  // Explicit opt-in from system classes
+  if (holder->trust_final_fields())
```

Comment:
This is definitely nicer than listing specific classes. It would be nicer again once we can make this exceptions go away.

Reply:
True, this occupies one of the 16 precious instance klass bits in runtime. I wish we can derive this from our final means final restrictions, but their setup is to permit use-sites to migrate more easily, and is harder for declaration sites to deduce if a declaration is easier to be permitted. We can consider blanket-trust when the JVM uses `--illegal-final-field-mutation=deny` without additional `--enable-final-field-mutation`.